### ISSUE TITLE
训练检测器，当一个epoch后中断，也不报错

### ISSUE THREAD

**Issue by swjtxw** [PaddlePaddle/PaddleOCR#2590]
[2021/04/22 13:58:03] root INFO: epoch: [1/1000], iter: 995, lr: 0.001000, loss: 2.760931, loss_shrink_maps: 1.510333, loss_threshold_maps: 0.913615, loss_binary_maps: 0.318070, reader_cost: 0.00000 s, batch_cost: 0.08692 s, samples: 5, ips: 11
.50463
[2021/04/22 13:58:03] root INFO: epoch: [1/1000], iter: 999, lr: 0.001000, loss: 3.015381, loss_shrink_maps: 1.626681, loss_threshold_maps: 0.926730, loss_binary_maps: 0.347289, reader_cost: 0.00000 s, batch_cost: 0.04986 s, samples: 4, ips: 16
.04399
`

**Comment by tink2123** [comment]
可以适当减小训练的batch_size 和 num_worker试试

**Comment by caopulan** [comment]
盲猜一手win10

**Comment by swjtxw** [comment]
> 盲猜一手win10

是的，是win10，请问怎么解决？

**Comment by swjtxw** [comment]
> 可以适当减小训练的batch_size 和 num_worker试试

我已经调到最小了，还是一样的

**Comment by ronaldpan** [comment]
参考 https://github.com/PaddlePaddle/Paddle/issues/32056

**Comment by Jane-Ding** [comment]
修改了tools/program.py 里的eval 方法，计数器 idx 是从0开始的，如果一共10张图片，最大idx=9，改了条件判断并放在最后：
if idx >= len(valid_dataloader)-1:
                break
可以运行了：
def eval(model, valid_dataloader, post_process_class, eval_class,
         use_srn=False):
    model.eval()
    with paddle.no_grad():
        total_frame = 0.0
        total_time = 0.0
        pbar = tqdm(total=len(valid_dataloader), desc='eval model:')
        for idx, batch in enumerate(valid_dataloader):
            # if idx >= len(valid_dataloader):
            #     break
            images = batch[0]
            start = time.time()
            if use_srn:
                others = batch[-4:]
                preds = model(images, others)
            else:
                preds = model(images)
            batch = [item.numpy() for item in batch]
            # Obtain usable results from post-processing methods
            post_result = post_process_class(preds, batch[1])
            total_time += time.time() - start
            # Evaluate the results of the current batch
            eval_class(post_result, batch)
            pbar.update(1)
            total_frame += len(images)
            print('idx ', idx)
            if idx >= len(valid_dataloader)-1:
                break
        # Get final metric，eg. acc or hmean
        metric = eval_class.get_metric()
        print('get metric: ', metric)
    pbar.close()
    model.train()
    metric['fps'] = total_frame / total_time
    return metric

**Comment by Jane-Ding** [comment]
并且把tools/program.py 里的train 方法，也改一下计数器，大概在201行，if idx >= len(train_dataloader)-1:
                break， 我在win10机器上就能正常训练，如果显存不大的话，把batch size调小，worker num调成1. 很奇怪的是，我在colab上都能正常训练。

**Comment by sml2h3** [comment]
找到个最靠谱的解决方法，不用改代码。
下载一个everything软件，然后搜索msvcp140.dll，按大小排序，把最大的那个msvcp140.dll替换掉C:\Windows\System32中的msvcp140.dll再启动项目就可以了，亲测可用
